nixos/modules/installer/tools/tools.nix

```diff
@@ -10,238 +10,292 @@ let
+              description = lib.mdDoc ''
+                the name of the package
+              '';
+            };
+
+            enable = lib.mkEnableOption (lib.mdDoc "the package as part of the installer tools.");
+
+            package = lib.mkOption {
+              type = lib.types.package;
+              description = lib.mdDoc ''
+                the derivation providing the package
+              '';
+            };
+          };
+
+          config.name = lib.mkDefault name;
+        }));
+      };
     };
   };
 
-  options.system.disableInstallerTools = mkOption {
-    internal = true;
-    type = types.bool;
-    default = false;
-    description = lib.mdDoc ''
-      Disable nixos-rebuild, nixos-generate-config, nixos-installer
-      and other NixOS tools. This is useful to shrink embedded,
-      read-only systems which are not expected to be rebuild or
-      reconfigure themselves. Use at your own risk!
-    '';
-  };
+  config =
+    let
+      installerTools = lib.filterAttrs (_: tool: tool.enable) config.system.installerTools.tools;
+    in
+    lib.mkIf (config.nix.enable && !config.system.disableInstallerTools) {
+
+      system.installerTools.tools = {
+        nixos-build-vms = {
+          enable = true;
+          package = makeProg {
+            name = "nixos-build-vms";
+            src = ./nixos-build-vms/nixos-build-vms.sh;
+            inherit (pkgs) runtimeShell;
+          };
+        };
+
+        nixos-enter = {
+          enable = true;
+          package = makeProg {
+            name = "nixos-enter";
+            src = ./nixos-enter.sh;
+            inherit (pkgs) runtimeShell;
+            path = makeBinPath [
+              pkgs.util-linuxMinimal
+            ];
+          };
+        };
+
+        nixos-generate-config = {
+          enable = true;
+          package = makeProg {
+            name = "nixos-generate-config";
+            src = ./nixos-generate-config.pl;
+            perl = "${pkgs.perl.withPackages (p: [ p.FileSlurp ])}/bin/perl";
+            hostPlatformSystem = pkgs.stdenv.hostPlatform.system;
+            detectvirt = "${config.systemd.package}/bin/systemd-detect-virt";
+            btrfs = "${pkgs.btrfs-progs}/bin/btrfs";
+            inherit (config.system.nixos-generate-config) configuration desktopConfiguration;
+            xserverEnabled = config.services.xserver.enable;
+          };
+        };
+
+        nixos-install = {
+          enable = true;
+          package = makeProg {
+            name = "nixos-install";
+            src = ./nixos-install.sh;
+            inherit (pkgs) runtimeShell;
+            nix = config.nix.package.out;
+            path = makeBinPath [
+              pkgs.jq
+              config.system.installerTools.tools.nixos-enter.package
+              pkgs.util-linuxMinimal
+            ];
+          };
+        };
+
+        nixos-option = {
+          enable = lib.versionOlder (lib.getVersion config.nix.package) "2.4pre";
```

Comment:
Since https://github.com/NixOS/nixpkgs/pull/237442, this should just be `true`

Reply:
Yep, I rebased the PR!

Reply:
might be github, but to me it still looks like the old code before #237442 (which switched this to always on).

Reply:
Hmm, weird, I could have sworn that I made the change during the rebase. I fixed it now in any case, thanks for the heads up!